task/06-conditions-n-loops-tasks.js

```diff
@@ -373,7 +479,54 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+    let strPath = "";
+   let first = pathes[0].split('/');
+   let second = pathes[1].split('/');
+   if (pathes.length > 2) {
+      let third = pathes[2].split('/');
+
+      for (let i = 0; i < second.length; i++) {
+         for(let j = 0; j < third.length; j++) {
+
+            first.filter((el) => {
+               if(el == second[i] && el == third[j])
+               strPath += el + '/' ;
+               return strPath;
+            })
+         }               
+      }
+      return strPath; 
+   }
+   for (let i = 0; i < second.length; i++) {
+   first.filter((el) => {
+      if(el == second[i])
+      strPath += el + '/' ;
+      return strPath;
+   })        
+  }
+  return strPath;
+
+//alternative way;
+//   let strPath = "";
+//   let first = pathes[0].split('/');
+//   let check = "";
+
+//   for (let i = 0; i < first.length; i++) {      
+//      check += first[i] + '/';
+//      let all = true;
+     
+//      for (let j = 0; j < pathes.length; j++) {                  
+//         if(!pathes[j].startsWith(check)) {            
+//            all = false;
+//         }                
+//      }
+//      if(all) {
+//         strPath = check; 
+//      }
+//      else break;
+      
+//   }   
+//   return strPath;  
```

Comment:
I think, your alternative solution is more readable. Anyway, both solutions are correct.